src/NServiceBus.Core/Persistence/InMemory/SagaPersister/InMemorySagaPersister.cs

```diff
@@ -3,203 +3,211 @@ namespace NServiceBus
     using System;
     using System.Collections.Concurrent;
     using System.Collections.Generic;
-    using System.Runtime.CompilerServices;
     using System.Threading.Tasks;
     using Extensibility;
-    using Persistence;
     using Sagas;
-
+    using Persistence;
+    
     class InMemorySagaPersister : ISagaPersister
     {
-        public Task Complete(IContainSagaData sagaData, SynchronizedStorageSession session, ContextBag context)
+        static readonly CorrelationId NoCorrelationId = new CorrelationId(typeof(object), "", new object());
+        readonly ConcurrentDictionary<Guid, Entry> _sagas = new ConcurrentDictionary<Guid, Entry>();
+        readonly ConcurrentDictionary<CorrelationId, Guid> _byCorrelationId = new ConcurrentDictionary<CorrelationId, Guid>();
```

Comment:
I think I might have messed up with R# conventions on my machine. We don't use `_` do we?